### PR TITLE
[FEATURE] Add Claude skills

### PR DESCRIPTION
**Pull request by albert-de-montserrat**
### Whats the purpose of this PR?
- [ ] Bug fix
- [x] New feature
- [ ] Documentation update
- [ ] Other, please explain

**Describe it in more detail below:**

Add a set of `SKILL.md` for Claude

**Checklist**
- [ ] The PR title is descriptive and starts with the appropriate tag: `[BUGFIX]`, `[ADDITION]`, `[DOC]`, etc.
- [ ] New tests (either assessing the correct behaviour of new internal functions or the correctness of a miniapps or benchmarks) were added, or old tests were updated
- [ ] Affected miniapps have also been updated
- [ ] The new feature was added in a way that does not break public API
- [ ] New documentation related to the new feature was added
- [ ] The new code follows the [contributor guidelines](https://github.com/PTsolvers/JustRelax.jl/blob/main/CONTRIBUTING.md), in particular the [Runic Style](https://github.com/fredrikekre/Runic.jl)

### CODE REVIEW

File: .claude/skills/backend-conventions/SKILL.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+- ParallelStencil's `@init_parallel_stencil` is **once per module per session**. Switching backend or dimensionality requires a fresh Julia process. Never try to re-init in the same session.
+- User-facing scripts/tests select the backend via `ENV["JULIA_JUSTRELAX_BACKEND"]` (`CPU`/`CUDA`/`AMDGPU`) and pass `backend_JR` (e.g. `CPUBackend`) to constructors like `StokesArrays(backend_JR, ni)`. JustPIC has its *own* backend constant (`JustPIC.CPUBackend`) — they are different types; don't mix them.
+- Dispatch on device is done via traits (`backend(::CuArray) = CUDABackendTrait()`), not via `isa CuArray` checks. Follow that pattern.
+- MPI/halo exchange goes through ImplicitGlobalGrid (`IGG`); CUDA-aware MPI is enabled in CI with `IGG_CUDAAWARE_MPI=1`.
+
+## Local environment
+
+This machine is macOS/Apple Silicon: only the CPU (Threads) backend runs locally. GPU correctness is verified by the CSCS GH200 CUDA pipeline (`ci/cscs-gh200.yml`). Write GPU-safe code (see [kernel-style](../kernel-style/SKILL.md)) and rely on CPU tests + CI.
```
Comment (aelligp) [inline .claude/skills/backend-conventions/SKILL.md]: While I'm fine with this currently, do we want to use a specific OS here? Theoretically it could check what backend is available e.g. through CUDA.functional() etc. If we want to test the GPU locally, we have enough resources in the offices to do so/

File: .claude/skills/kernel-style/SKILL.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+description: ParallelStencil kernel idioms and GPU-compatibility rules for JustRelax.jl, plus the Runic formatting requirement. Use when writing or editing compute kernels or any Julia source.
+---
+
+# Kernel style
+
+## ParallelStencil idioms
+
+- Two kernel forms:
+  - `@parallel function f!(A, B, ...)` with `@all(A) = ...`, `@inn`, `@av`, `@d_xa`, etc. (FiniteDifferences2D/3D macros) for pure stencil updates.
+  - `@parallel_indices (i, j) function f!(...)` when you need explicit indexing, branching per cell, or CellArray access. Always `return nothing` at the end.
```
Comment (aelligp) [inline .claude/skills/kernel-style/SKILL.md]:
```suggestion
  - `@parallel_indices (I...) function f!(...)` when you need explicit indexing, branching per cell, or CellArray access. Always `return nothing` at the end. `I...` is the dimension agnostic way if it is a general kernel. Use `i, j` or `i, j, k` for dimension specific kernels  
```

File: .claude/skills/miniapps-and-benchmarks/SKILL.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+## Running one
+
+```bash
+cd miniapps
+julia --project=. -e 'using Pkg; Pkg.develop(path=".."); Pkg.instantiate()'   # first time, to run against local JustRelax
+JULIA_JUSTRELAX_BACKEND=CPU julia --project=. benchmarks/stokes2D/<case>/<script>.jl
+```
+
+Notes:
+- Scripts choose the backend the same way tests do (env var / editing the `backend_JR` constant at the top). On this machine only CPU works.
```
Comment (aelligp) [inline .claude/skills/miniapps-and-benchmarks/SKILL.md:27]: Same issue with the backend as in the comment above

File: .claude/skills/release/SKILL.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+---
+
+# Releasing
+
+JustRelax.jl is registered in the Julia General registry. Releases are driven from `main`.
+
+## Steps
+
+1. **Bump `version`** in `Project.toml` (currently `0.x` — patch for fixes, minor for features/breaking, per common 0.x SemVer practice in the Julia ecosystem).
+2. Make sure `[compat]` entries are current. The tightly-coupled companion packages are **GeoParams** and **JustPIC** (both PTsolvers); a JustRelax release often follows a compat bump for one of these. Dependabot handles GitHub Actions, not Julia compat — compat bumps are manual or via CompatHelper-style PRs.
```
Comment (aelligp) [inline .claude/skills/release/SKILL.md:13]: Dependabot also handles Julia compass now IIRC

File: .claude/skills/release/SKILL.md
Comment (aelligp) [inline .claude/skills/release/SKILL.md:1]: Are you sure you want to add this skill? Not sure I want to give Claude the capability to release JR versions

File: .claude/skills/running-tests/SKILL.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+How `test/runtests.jl` works:
+- Only files named `test_*.jl` are collected. `deprecated_*.jl` files are skipped.
+- Non-MPI tests run in parallel via **ParallelTestRunner**; a worker is spawned per test except for a hardcoded list of light tests (traits, types, conversions, mask, mini kernels, interpolations, boundary conditions) that run without a worker.
+- Files with `MPI` in the name (`test_*_MPI.jl`) are split out and run sequentially with `mpiexec -n 2`.
+- You can filter: `Pkg.test(test_args=["test_diffusion2D"])` runs only matching tests (burstedde and VanKeken are excluded when filtering).
+
+## Practical guidance
+
+- The full suite is **slow** (solver tests run actual pseudo-transient iterations). Prefer one targeted test file. Quick smoke tests: `test_types`, `test_traits`, `test_arrays_conversions`, `test_boundary_conditions2D` (seconds, no solver). Physics regressions: `test_shearband2D`, `test_diffusion2D`, `test_Blankenbach`, `test_VanKeken` (minutes).
```
Comment (aelligp) [inline .claude/skills/running-tests/SKILL.md]: `test_VanKeken` is skipped right now

File: .claude/skills/running-tests/SKILL.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+- MPI tests locally: `mpiexec -n 2 julia --project=. --startup-file=no test/test_diffusion2D_multiphase_MPI.jl` (use the MPI.jl-provided `mpiexecjl` if no system MPI is configured).
+- GPU backends cannot be tested on this machine (macOS/Apple Silicon); CPU-pass + CI (CSCS GH200 pipeline in `ci/cscs-gh200.yml` runs CUDA + MPI tests) is the verification path.
```
Comment (aelligp) [inline .claude/skills/running-tests/SKILL.md]: macOS again

**Review by aelligp — CHANGES_REQUESTED** [review]
Looks ok, a few minor comments and one decision we need to make with the release skill

File: .claude/skills/release/SKILL.md
Comment (albert-de-montserrat) [reply .claude/skills/release/SKILL.md:1]: A release is nothing but updting the version and making a PR. This skill needs to be activated by us and it's just a lazy alternative.

Comment (aelligp) [reply .claude/skills/release/SKILL.md:1]: Fine with me then
